psst-gui/src/token_utils.rs

```diff
@@ -0,0 +1,197 @@
+
+impl TokenUtils {
+    #[inline]
+    fn set_bearer(session: &SessionService, token: Option<&str>) {
+        match token {
+            Some(t) => {
+                let s = t.to_string();
+                session.set_oauth_bearer(Some(s.clone()));
+                WebApi::global().set_oauth_bearer(Some(s));
+            }
+            None => {
+                session.set_oauth_bearer(None);
+                WebApi::global().set_oauth_bearer(None);
+            }
+        }
+    }
+
+    #[inline]
+    fn set_refresh(session: &SessionService, token: Option<&str>) {
+        match token {
+            Some(t) => {
+                let s = t.to_string();
+                session.set_oauth_refresh_token(Some(s.clone()));
+                WebApi::global().set_oauth_refresh_token(Some(s));
+            }
+            None => {
+                session.set_oauth_refresh_token(None);
+                WebApi::global().set_oauth_refresh_token(None);
+            }
+        }
+    }
+
+    /// Apply tokens to runtime holders (Session and WebApi).
+    /// - If `access_token` is Some, set it on both Session and WebApi. If None,
+    ///   clear on both.
+    /// - If `refresh_token` is Some, set it on both. If None, do not change
+    ///   existing refresh tokens unless `clear_refresh_if_none` is true (then
+    ///   it will be cleared).
+    pub fn apply_runtime_tokens(
+        session: &SessionService,
+        access_token: Option<&str>,
+        refresh_token: Option<&str>,
+        clear_refresh_if_none: bool,
+    ) {
+        // Summarize action in a single log line
+        let access_state = if access_token.is_some() {
+            "set"
+        } else {
+            "cleared"
+        };
+        let refresh_state = if refresh_token.is_some() {
+            "set"
+        } else if clear_refresh_if_none {
+            "cleared"
+        } else {
+            "unchanged"
+        };
+        log::info!("TokenUtils: runtime access={access_state}, refresh={refresh_state}");
+
+        // Apply access token on both holders
+        Self::set_bearer(session, access_token);
+
+        // Apply or clear refresh token as requested
+        if let Some(rt) = refresh_token {
+            Self::set_refresh(session, Some(rt));
+        } else if clear_refresh_if_none {
+            Self::set_refresh(session, None);
+        }
+    }
+
+    /// Persist tokens into Config (and optionally save).
+    /// - If `access_token` is Some, store it. If None, clear it.
+    /// - If `refresh_token` is Some, store it. If None, do not change the
+    ///   persisted refresh token unless `clear_refresh_if_none` is true (then
+    ///   it will be cleared).
+    pub fn persist_tokens(
+        config: &mut Config,
+        access_token: Option<&str>,
+        refresh_token: Option<&str>,
+        clear_refresh_if_none: bool,
+        save: bool,
+    ) {
+        log::info!("TokenUtils: persist save={save} clear_refresh_if_none={clear_refresh_if_none}");
```

Comment:
All of these TokenUtil logs seem better suited to debug, right? I guess it is useful information to see when the token is refreshed. Aside: how do I change the logging level when running the binary?